pkg/crc/cluster/cluster.go

```diff
@@ -22,21 +22,44 @@ func WaitForSsh(driver drivers.Driver) error {
 	return errors.RetryAfter(10, checkSshConnectivity, time.Second)
 }
 
-// CheckCertsValidity checks if the cluster certs have expired
-func CheckCertsValidity(driver drivers.Driver) error {
+// CheckCertsValidityUsingBundleBuildTime check if the cluster certs going to expire soon.
+func CheckCertsValidityUsingBundleBuildTime(buildTime time.Time) bool {
+	certExpiryDate := buildTime.AddDate(0, 1, 0)
+	// Warn user if the cert expiry going to happen in 7 days
+	timeAfter7Days := time.Now().AddDate(0, 0, 7)
+	return timeAfter7Days.After(certExpiryDate)
```

Comment:
This will start to return false from the _starting_ of the 7 days.. maybe mention that in the comment.